### PR TITLE
adds option to extract barcode from read2 only

### PR DESCRIPTION
**Pull request by TomSmithCGAT**
This adds the option `--read2-only` for `extract` and `whitelist` to only extract barcodes from read2.

See #522 for motivation

### CODE REVIEW

**Review by IanSudbery — COMMENTED** [review]
I don't see any immediate problems, and I assume that you have run the tests. There was one question I asked. 

Also, what happens if the mode is string rather than regex, is that covered? And documented?

File: umi_tools/extract_methods.py [hunk 1]
```diff
@@ -542,7 +532,7 @@ def __call__(self, read1, read2=None):
         # Check for regex is more complex.
         # Reads too short with regex pattern will silently fail to match
         if self.method == "string":
-            if(len(read1.seq) < len(self.pattern)):
+            if(self.pattern and (len(read1.seq) < len(self.pattern))):
```
[the remark this answers is not in the document]
Comment (TomSmithCGAT) [reply umi_tools/extract_methods.py:535]: Yes. Good point. I'll add that

**Comment by TomSmithCGAT** [comment]
Yep. Tests all run OK and new tests include mode=string and mode=regex. 

By 'And documented?', do you mean is the new option to just use read2 only documented? If so, yes, usual documentation in the tool, plus FAQs.

File: umi_tools/extract_methods.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (TomSmithCGAT) [reply umi_tools/extract_methods.py:535]: Ah, it's already there! See line 540

**Pull request merged**